### ISSUE TITLE
Error with Cloudflare DNS "could not find the start of authority" (but works fine in lego cli)

### ISSUE THREAD

**Issue by dvasdekis** [vancluever/terraform-provider-acme#453]
Firstly, thank you for making this awesome provider! I'm fairly certain something is going wrong within this provider, rather than lego, as lego works fine:

`lego --email "my-email@REDACTED.com" --dns cloudflare -d '*.REDACTED.com' -d "REDACTED.com" --accept-tos --dns.resolvers="1.1.1.1" --server "https://acme-staging-v02.api.letsencrypt.org/directory" --pfx run`

However with the below config, 

```hcl
provider "acme" {
  server_url = "https://acme-staging-v02.api.letsencrypt.org/directory"
}

resource "acme_certificate" "cert" {
  account_key_pem          = acme_registration.my-registration.account_key_pem
  common_name              = "${data.cloudflare_zone.zone.name}"
  subject_alternative_names = ["*.${data.cloudflare_zone.zone.name}"]
  recursive_nameservers = ["1.1.1.1:53"]  # Cloudflare nameserver
  certificate_p12_password = random_password.my-cert.result
  min_days_remaining = 88
  dns_challenge {
    provider = "cloudflare"
    config = {
      CLOUDFLARE_DNS_API_TOKEN=var.cloudflare_api_token
    }
  }
}
```

I get
```
  │ Error: error creating certificate: error: one or more domains had a problem:
  │ [*.REDACTED.com] [*.REDACTED.com] acme: error presenting token: 2 errors occurred:
  │ 	* rpc error: code = Unknown desc = cloudflare: could not find zone for domain "REDACTED.com": [fqdn=_acme-challenge.REDACTED.com.] could not find the start of authority for '_acme-challenge.REDACTED.com.': DNS call error: read udp 10.195.175.100:56056->1.1.1.1:53: i/o timeout [ns=1.1.1.1:53, question='au. IN  SOA']
  │ 	* error encountered while presenting token for DNS challenge: rpc error: code = Unknown desc = cloudflare: could not find zone for domain "REDACTED.com": [fqdn=_acme-challenge.REDACTED.com.] could not find the start of authority for '_acme-challenge.REDACTED.com.': DNS call error: read udp 10.195.175.100:56056->1.1.1.1:53: i/o timeout [ns=1.1.1.1:53, question='au. IN  SOA']
```
Further, when I check cloudflare with this same token using the CLI, I can see the _acme-challenge TXT records created. But during apply with the Terraform provider, there are no TXT records, and no DEBUG entries to show that Terraform is communicating with cloudflare.

[It could be a duplicate of this issue](https://github.com/vancluever/terraform-provider-acme/issues/376), but I am very happy to hang around and help! However, rolling back to 2.8.0 doesn't solve the problem.

# Versions:
lego cli: 4.20.2
Terraform: 1.9.8 (tried on windows_amd64 and linux_amd64)
acme_provider: v2.28.0

**Comment by vancluever** [comment]
@dvasdekis sorry for the delay in replying to this one!

Unfortunately I don't have a Cloudflare account over here, so this one will be hard for me to reproduce, but this part is interesting to me:

> `DNS call error: read udp 10.195.175.100:56056->1.1.1.1:53: i/o timeout [ns=1.1.1.1:53, question='au. IN  SOA']`

Is there anything that could be affecting your ability to communicate with the Cloudflare DNS server?

**Comment by dvasdekis** [comment]
I've spent some time today trying a variety of DNS providers, and the DNS issue looks like a real problem, except for the base CLI client being able to happily use the `1.1.1.1` DNS resolver (whereas Terraform can't on the same machine). For this reason I don't think it's a DNS access issue.

Is there some way I can see, within Terraform logs, how this driver is interacting with Cloudflare?

**Comment by vancluever** [comment]
@dvasdekis not within logs unfortunately, but you can see how the plugin process on our side plumbs through to [`Present`](https://github.com/vancluever/terraform-provider-acme/blob/086fd3773834f4b5b8b6135a0ea580a850962a02/acme/dnsplugin/plugin.go#L94-L97), which ends up on the lego plugin side [here](https://github.com/go-acme/lego/blob/87b7e7191f52db48722de068eaf519b95e728ed6/providers/dns/cloudflare/cloudflare.go#L153-L157).

**Comment by dvasdekis** [comment]
Perhaps that could be the feature request then? It looks like the parameters are getting passed with no transformations or significant logic, so I don't know how to dig into this further without logs of the `dns_challenge` block. Would that be a big deal?

**Comment by vancluever** [comment]
@dvasdekis I don't think that would really give us much in the way of detail, unfortunately.

`Present` is part of the lego DNS provider API and we are not doing anything else other than wrapping that call so that multiple provider instances can hold different settings (doesn't necessarily apply to you unless you are doing multiple certificates in the same configuration). By the time it's hit in that call, you're already deep enough into the lego stack that it's reaching out to DNS plugins.

The fact that you're seeing the error message `could not find zone for domain "REDACTED.com"` means that the data has made it to `Present` in the DNS provider side okay, as `Present` only takes the domain, token, and key. The domain you're seeing in the error message (from the plugin), the key is your TXT record value, and the token is for the mutex and isn't playing into the process here.

**Comment by vancluever** [comment]
@dvasdekis closing this due to age. Let me know if you have any new details to share!
